packages/react-core/src/components/Alert/Alert.tsx

```diff
@@ -80,6 +88,9 @@ export const Alert: React.FunctionComponent<AlertProps & OUIAProps> = ({
         {action && (typeof action === 'object' || typeof action === 'string') && (
```

Comment:
Perhaps a better API would be to leave `action`'s type as `React.ReactNode` and use `React.cloneElement` to add the `title` and `variantLabel` props? This way class components could be passed as well, and we don't have to amend the API.

```tsx
{React.isValidElement(action) && (
  <div className={css(styles.alertAction)}>
    {React.cloneElement(action as React.ReactElement, {
      title,
      variantLabel
    })}
  </div>
)}
```

Consumers would then be able to use added `title` and `variantLabel` props like this:

```tsx
<Alert action={({title, variantLabel}) => <div>{title}{variantLabel}</div>} />
```

OR like this:

```tsx
class MyAlertAction extends React.Component {
  render() {
    return <div>{this.props.title}{this.props.variantLabel}</div>;
  }
}

<Alert action={<MyAction />} />
```